### DISCUSSION TITLE
How to get the object's size?

### DISCUSSION THREAD

**Discussion by bidzhao** [CadQuery/cadquery#1223] [Q&A] [answered]
Hello,

I want to scale the imported model according to its size. But I wonder how to get the objects' size after importing from step/dxf files?

For example:

model = cq.importers.importStep("model.stp")
for o in model.objects:
    # how to get the object's size in (x, y, z)?

**Comment by jmwright** [comment, marked as answer]
You can get the size of an object via the BoundingBox in this way.

```python
import cadquery as cq

result = cq.Workplane().box(10, 20, 30)
cq.exporters.export(result, "model.step")

model = cq.importers.importStep("model.step")

for o in model.objects:
    log(o.BoundingBox().xlen)  # prints 10.0
    log(o.BoundingBox().ylen)  # prints 20.0
    log(o.BoundingBox().zlen)  # prints 30.0
```

**Reply by bidzhao** [reply]
Thank you! It works perfectly.

**Reply by adam-urbanczyk** [reply]
On some objects you can also use `Surface`, `Volume` and `Length` methods.
